File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/reddit/RedditSubComponent.java
package de.jbamberger.offlineclient.di.reddit;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import de.jbamberger.offlineclient.ui.reddit.RedditActivity;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditScope
@Subcomponent(modules = {RedditModule.class})
public interface RedditSubComponent extends AndroidInjector<RedditActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<RedditActivity> {
    }
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/reddit/RedditSubComponent.kt
package de.jbamberger.offlineclient.di.reddit

import dagger.Subcomponent
import dagger.android.AndroidInjector
import de.jbamberger.offlineclient.ui.reddit.RedditActivity

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditScope
@Subcomponent(modules = [RedditModule::class])
interface RedditSubComponent : AndroidInjector<RedditActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<RedditActivity>()
}
